ECMA2Yaml/ECMAHelper/Models/ECMAStore.cs

```diff
@@ -314,26 +314,37 @@ private void BuildAssemblyMonikerMapping(ReflectionItem item)
 
         private void BuildExtensionMethods()
         {
-            if (_extensionMethods == null || _extensionMethods.Count == 0)
-            {
-                return;
-            }
-            ExtensionMethodsByMemberDocId = _extensionMethods.ToDictionary(ex => ex.MemberDocId);
-
+            _extensionMethods = new List<Member>();
             foreach (var m in MembersByUid.Values)
             {
-                if (!string.IsNullOrEmpty(m.DocId) && ExtensionMethodsByMemberDocId.ContainsKey(m.DocId))
+                if (!string.IsNullOrEmpty(m.DocId))
                 {
-                    m.IsExtensionMethod = true;
-                    ExtensionMethodsByMemberDocId[m.DocId].Uid = m.Uid;
-                    ExtensionMethodsByMemberDocId[m.DocId].ParentType = m.Parent;
+                    if (m.Parameters != null && m.Parameters.Where(p => { return p.RefType == "this"; }).Count() > 0)
```

Comment:
we can save it as `var thisParam = m.Parameters.FirstOrDefault(p => { return p.RefType == "this"; });` and use it in both if conditions down below.